File: src/main/java/testingil/webinar/tddapi/s17/Navigator.java
package testingil.webinar.tddapi.s17;

public class Navigator {

	private DistanceProvider distProvider;
	private Location startingPoint;
	private Location endPoint;

	public Navigator(Location startingPoint, DistanceProvider distProvider) {
		this.distProvider = distProvider;
		this.startingPoint = startingPoint;
	}

	public Distance getDistanceFromDestination()	{
		return distProvider.getDistance(startingPoint, endPoint);
	}

	public void setDestination(Location dest) {
		this.endPoint = dest;
	}

}

File: C#/TddApi/NavigationApp/s17.fill_controller/Navigator.cs
using System;

namespace NavigationApp.s17.fill_controller
{
    public class Navigator
    {
        DistanceProvider distProvider;
        Location startingPoint;
        Location destination;

        public Navigator(Location loc, DistanceProvider distService)
        {
            distProvider = distService;
            startingPoint = loc;
        }

        public Distance getDistanceFromDestination()
        {
            return distProvider.getDistance(startingPoint, destination);
        }

        public void setDestination(Location dest)
        {
            destination = dest;
        }
    }
}